src/main/java/ch/njol/skript/expressions/ExprEnchantmentLevel.java

```diff
@@ -1,95 +1,140 @@
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.EnchantmentType;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Enchantment Level")
+@Description("The level of a particular <a href='classes.html#enchantment'>enchantment</a> on an item")
+@Examples({"player' tool is a sword of sharpness:",
+		"	message \"You have a sword of sharpness %level of sharpness of the player's tool% equipped\""})
+@Since("2.0")
+public class ExprEnchantmentLevel extends SimpleExpression<Integer> {
+	
+	static {
+		Skript.registerExpression(ExprEnchantmentLevel.class, Integer.class, ExpressionType.PROPERTY,
+				"[the] [enchant[ment]] level of %enchantments% (on|of) %itemtypes%",
+				"[the] %enchantments% [enchant[ment]] level (on|of) %itemtypes%",
+				"%itemtypes%'[s] %enchantments% [enchant[ment]] level",
+				"%itemtypes%'[s] [enchant[ment]] level of %enchantments%");
+	}
+	
+	@SuppressWarnings("null")
+	private Expression<ItemType> items;
+	@SuppressWarnings("null")
+	private Expression<Enchantment> enchants;
+	
+	@SuppressWarnings({"unchecked", "null"})
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		int i = matchedPattern < 2 ? 1 : 0;
+		items = (Expression<ItemType>) exprs[i];
+		enchants = (Expression<Enchantment>) exprs[i ^ 1];
+		return true;
+	}
+	
+	@Override
+	protected Integer[] get(final Event e) {
+		Enchantment[] enchantments = enchants.getArray(e);
+		return Stream.of(items.getArray(e))
+			.map(ItemType::getEnchantmentTypes)
+			.flatMap(Stream::of)
+			.filter(enchantment -> CollectionUtils.contains(enchantments, enchantment.getType()))
+			.map(EnchantmentType::getLevel)
+			.toArray(Integer[]::new);
+	}
+	
+	@Override
+	public boolean isSingle() {
+		return items.isSingle() && enchants.isSingle();
+	}
+	
+	@Override
+	public Class<? extends Integer> getReturnType() {
+		return Integer.class;
+	}
+	
+	@Nullable
+	@Override
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		switch (mode) {
+			case SET:
+			case REMOVE:
+			case ADD:
+				return CollectionUtils.array(Number.class);
+			default:
+				return null;
+		}
+	}
+	
+	@Override
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
+		if (delta == null)
+			return;
+		
+		ItemType[] source = items.getArray(e);
+		Enchantment[] enchantments = enchants.getArray(e);
+		int newLevel = ((Number) delta[0]).intValue();
+		
+		if (mode == ChangeMode.ADD)
+				newLevel += 1;
+		else if (mode == ChangeMode.REMOVE)
+				newLevel -= 1;
```

Comment:
When you have an item with, say, level 3 of some enchantment, and try to add, say, 5 to it, won't it set it to 6?
Because `int newLevel = ((Number) delta[0]).intValue();` - newLevel is 5 right?
`newLevel += 1;` - and now it's 6

Same thing with removing, it'd be 4